src/doc/src/reference/build-scripts.md

```diff
@@ -313,13 +314,18 @@ link-arg=FLAG` option][link-arg] to the compiler, but only when building a
 `cdylib` library target. Its usage is highly platform specific. It is useful
 to set the shared library version or the runtime-path.
 
-### `cargo::warning=MESSAGE` {#cargo-warning}
+### `cargo::error=MESSAGE` {#cargo-error}
 
-The `warning` instruction tells Cargo to display a warning after the build
-script has finished running. Warnings are only shown for `path` dependencies
-(that is, those you're working on locally), so for example warnings printed
+The `error` instruction tells Cargo to display an error after the build
+script has finished running. Errors are only shown for `path` dependencies
+(that is, those you're working on locally), so for example errors printed
 out in [crates.io] crates are not emitted by default. The `-vv` "very verbose"
-flag may be used to have Cargo display warnings for all crates.
+flag may be used to have Cargo display errors for all crates.
+
+### `cargo::warning=MESSAGE` {#cargo-warning}
+
+The `warning` instruction is just like [`error`](#cargo-error), except that it
+will output warnings.
```

Comment:
This does not match what the cargo team accepted in https://github.com/rust-lang/cargo/issues/10159#issuecomment-988104375

`cargo::error` should force the build to fail just like a bad exit code, independent of where it is in the dependency graph.  This means we should always print a `cargo::error`, even for transitive dependencies.

Reply:
My thinking was to leave this for now, as there was a discussion about this later in the issue. But I can change it.

Reply:
I don't think a feature flag makes a lot of sense here, as a library that emits `cargo::error` would have to require that any client app runs with the feature flag enabled. I'll go with alternative 1.